### PR TITLE
feat(settings): New banner component with updated styling

### PR DESCRIPTION
**Pull request by vpomerleau**
## Because

* We are updating the design of the Banner component
* We are updating the design of the AlertBar component

## This pull request

* Add new `Banner` component that replaces the old one and add stories, tests, l10n
* Add new icon components, minify all icons
* Replace Image used in `SigninTokenCode`, `ConfirmSignupCode`, `SigninUnblock` (use `EmailCodeImage` instead of `MailImage` which referenced an emailed link)
* Update `alertBar` component, add stories, update tests

## Issue that this pull request solves

Closes: #FXA-9714

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Other information (Optional)

Figma design for updated/new components:
[AlertBar](https://storage.googleapis.com/mozilla-storybooks-fxa/commits/7d9199066ff8f76bb2ede25f9b88f88321df0b70/fxa-settings/index.html?path=/story/components-alertbar--error)

[BannerNew](https://storage.googleapis.com/mozilla-storybooks-fxa/commits/7d9199066ff8f76bb2ede25f9b88f88321df0b70/fxa-settings/index.html?path=/story/components-bannernew--resend-code-banner)

UX Review ticket: https://mozilla-hub.atlassian.net/browse/FXA-10683

### CODE REVIEW

File: packages/functional-tests/pages/resetPassword.ts
Comment (vpomerleau) [inline packages/functional-tests/pages/resetPassword.ts:74]: Accepting a looser match here because the success message includes a period when shown in a banner but not in alertBar

File: packages/fxa-settings/src/components/BannerNew/index.stories.tsx
Comment (vpomerleau) [inline packages/fxa-settings/src/components/BannerNew/index.stories.tsx:12]: If we could use the Storybook controls panel, we would be able to significantly cut down on the number of stories needed to show all the possible states.

File: packages/fxa-settings/src/components/ConfirmWithLink/en.ftl
Comment (vpomerleau) [inline packages/fxa-settings/src/components/ConfirmWithLink/en.ftl:1]: The `ConfirmWithLink` component no longer seems to be used anywhere.

File: packages/fxa-settings/src/pages/Signin/SigninTokenCode/index.tsx
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/Signin/SigninTokenCode/index.tsx:109]: Should we check if the error is an incorrect/invalid code and if so display that in a tooltip (to also return focus to the input with error styling), and display every other error in a banner (vs only displaying the throttling errors in a banner)?

**Review by bcolsson — APPROVED** [review]

File: packages/fxa-settings/src/pages/InlineRecoveryKeySetup/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/InlineRecoveryKeySetup/index.tsx:71]: Much nicer 🧹⭐

I didn't go through and check all of the margin spacing on these fwiw (like `mt-0` being passed in here).

File: packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/container.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/container.tsx:101]: Do we have to type cast this?

File: packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/container.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/container.tsx:99]: Do you know when we use "sign-in" vs "sign in"? I just assumed we should never have the dash, but now I'm not sure?

File: packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/container.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/container.tsx:147]: Same here, preferably we don't have any type casting.

File: packages/fxa-settings/src/pages/Signin/SigninUnblock/index.test.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/Signin/SigninUnblock/index.test.tsx:117]: Do we want to move away from "Illustration"?

File: packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/container.test.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/container.test.tsx:282]: `expect`/`toBeInTheDocument();` can be removed here too

**Review by LZoog — APPROVED** [review]
Nice improvements @vpomerleau!!

I know I left a lot of comments but I'm going to r+ since I don't really see any that I'd consider _blocking_, I'll let you decide what you want to update here and feel free to LMK if you want me to do another pass here before merge. 🙂

File: packages/fxa-settings/src/components/BannerNew/index.stories.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/components/BannerNew/index.stories.tsx:12]: Aw thanks ❤️ 

It's not currently working because our storybook package versions are not aligned and there are conflicts that prevent the add-on from working. I spent some time looking at upgrading, but it was a rabbit role. It'll have to be done separately.

Good news is that both CoPilot and ChatGPT seem to do a good job with the tedious task of converting the story format when we get to that point.

File: packages/fxa-settings/src/components/BannerNew/index.test.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/components/BannerNew/index.test.tsx:69]: I thought so too, but I've had some odd cases where tests interfered with each other when I used `userEvent`, even when set up in each individual test. I'm a bit suspicious and need to spend some time reading up on it more.

File: packages/fxa-settings/src/components/BannerNew/index.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/components/BannerNew/index.tsx:52]: 👍 I'll go with option 2, I want to make sure that one of the two is always included.

File: packages/fxa-settings/src/components/BannerNew/index.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/components/BannerNew/index.tsx:73]: Agree, blue on red is always a bit hard on the eyes, but our link colour does indeed meet AA level on all banner backgrounds.

File: packages/fxa-settings/src/components/BannerNew/interfaces.ts
Comment (vpomerleau) [reply packages/fxa-settings/src/components/BannerNew/interfaces.ts:23]: Right, I went with `interfaces.ts` for consistency, but didn't see a need to use interfaces here (based on https://www.totaltypescript.com/type-vs-interface-which-should-you-use)

File: packages/fxa-settings/src/components/Settings/AlertBar/index.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/components/Settings/AlertBar/index.tsx:35]: This was accidental, I'll add them back in. Thank you!

File: packages/fxa-settings/src/components/Settings/AlertBar/index.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/components/Settings/AlertBar/index.tsx:112]: It was causing some trouble with the test mocks and didn't seem worth the battle for one string.

File: packages/fxa-settings/src/models/AlertBarInfo.ts
Comment (vpomerleau) [reply packages/fxa-settings/src/models/AlertBarInfo.ts:4]: For this use case, I don't see an advantage to using an enum vs a union type (https://www.typescriptcourse.com/string-literal-unions-over-enums), though we could perhaps use a shared type?

File: packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/container.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/container.tsx:101]: We don't _have_ to cast it, but it helped me spot errors when I initially misnamed one of the keys.

File: packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/container.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/container.tsx:99]: `sign in` is the verb, and `signin` or `sign-in` (both are  accepted) are nouns or adjectives.

File: packages/fxa-settings/src/pages/Signin/SigninUnblock/index.test.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/Signin/SigninUnblock/index.test.tsx:117]: We do, and my PR was initially even bigger because I had also updated all the image aria-labels... I opted to back out those changes after rebasing off the other two recent PRs that updated images/icons. I can file a ticket.

The reason for the switch here is that I replaced the image used on this page (we still had the email link image even though this an email code) but did not make changes to the images themselves.

File: packages/fxa-settings/src/components/BannerNew/index.test.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/components/BannerNew/index.test.tsx:24]: This one is a matter of opinion, but I tend to prefer making the assertion explicit.
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-get-variants-as-assertions

Comment (LZoog) [reply packages/fxa-settings/src/components/BannerNew/index.test.tsx:24]: Interesting, I guess I feel like this just double tests it and adds extra characters/lines to files. I wonder if there is a test execution time difference. (If the team prefers your way, that's OK by me, might just be good to establish which way we want to go moving forward)

File: packages/fxa-settings/src/components/BannerNew/interfaces.ts
Comment (LZoog) [reply packages/fxa-settings/src/components/BannerNew/interfaces.ts:23]: I guess this is another thing we could align on as a team. I don't have a strong preference but I thought I remembered Julian saying we should always prefer interfaces where possible 😅 Wondering now if I got that backwards?

File: packages/fxa-settings/src/models/AlertBarInfo.ts
Comment (LZoog) [reply packages/fxa-settings/src/models/AlertBarInfo.ts:4]: Shared type WFM!

File: packages/fxa-settings/src/components/BannerNew/index.test.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/components/BannerNew/index.test.tsx]: Updated.

File: packages/fxa-settings/src/components/Icons/index.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/components/Icons/index.tsx:7]: I think they mostly were, but I'll run them through again and rename to `min.svg` for clarity.

**Pull request merged**
